### ISSUE TITLE
Trouble validating with oneOf

### ISSUE THREAD

**Issue by fernandesPabloVexter** [express-validator/express-validator#753]
I'm trying to validate one field or another, based on the value i get through params.
My code is as follows:

```
//validation.mid
const { body, oneOf, param } = require('express-validator');

exports.validate = () => {
    return [
        oneOf([
            param('emailType')
                .if((value, { req }) => {
                    value === 'from'
                }),
            body('from').exists().isEmail().withMessage('Invalid from email.')
        ], [
                param('emailType')
                    .if((value, { req }) => {
                        value === 'to'
                    }),
                body('to').exists().isEmail().withMessage('Invalid to email.')
            ])]
}
```

And the `index.route`:
```
const controller = require('./src/email.controller');
const validation = require('./src/validation.middleware');

module.exports = function (app) {

    app.route('/:emailType')
        .post(validation.validate(), controller.sendEmail)
}
```
**With this validation, nothings throws an error, doesn't matter if i send "from" or "to" or the value inside the body.**

I have also tried using `custom`:
```
const { body, oneOf, param } = require('express-validator');

exports.validate = () => {
    return [
        oneOf([
            param('emailType')
                .custom((value, { req }) => {
                    if (value != 'from') {
                        return false;
                    }
                    return true;
                }),
            body('from').exists().isEmail().withMessage('From errado.')
        ], [
                param('emailType')
                    .custom((value, { req }) => {
                        if (value != 'to') {
                            return false;
                        }
                        return true;
                    }),
                body('to').exists().isEmail().withMessage('To errado.')
            ])]

}
```
Using Custom, if i send the param "**from**" and in the body (`from` field) a value other than an e-mail, it throws the error:
```
{
    "errors": [
        {
            "msg": [
                null,
                null
            ],
            "param": "_error",
            "nestedErrors": [
                {
                    "value": "to",
                    "msg": "Invalid value",
                    "param": "emailType",
                    "location": "params"
                },
                {
                    "value": "marccom.br",
                    "msg": "From errado.",
                    "param": "from",
                    "location": "body"
                }
            ]
        }
    ]
}
```
If i send the param "**to**" in the body, `to` field, the `from` field gets validated again, instead of the `to` field.

So, to summarize: Using Custom and sending either "**from**" and "**to**" params will validate the `from` field in the body, using if nothings gets validated and the error object returned brings more errors than it should.
Can anyone, please, point what am i doing wrong in this code?
Thanks in advance.

**Comment by gustavohenke** [comment]
> using if nothings gets validated

That's because you're missing `return`s in the example using `if` 🙂 

<img width="612" alt="Screen Shot 2019-08-05 at 09 00 18" src="https://user-images.githubusercontent.com/826553/62430376-843ee000-b75f-11e9-9129-34e160a4ca66.png">
